org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/GuardedPattern.java

```diff
@@ -27,121 +27,82 @@
 
 public class GuardedPattern extends Pattern {
 
-	public Pattern[] patterns;
+	public Pattern primaryPattern;
```

Comment:
I don't think having the guard connected to only one pattern is a good idea. When working with case statements, the user of the AST needs to know that they should check the latest pattern in the list of patterns for a guard, and will need to proceed differently based on that result.

Ideally, we should have the guard as a child of the case statement to match up with how the spec is written.

The reason I placed the list of patterns as a child of GuardedPattern instead of under CaseStatement was to avoid deprecating GuardedPattern and reduce the amount of changes I needed to do to code generation. Maybe this was a bad idea, since code generation needed to be rewritten anyways.